api/checkTokenMiddleware.ts

```diff
@@ -0,0 +1,22 @@
+import axios from "axios";
+import jwt from "jwt-decode";
+import { store } from "../store/store";
+import { LogOut } from "../store/userReducer";
+
+export async function checkToken() {
+  const token = localStorage.getItem("access_token");
+  const data = jwt(token || "");
+  const refresh_token = localStorage.getItem("refresh_token");
+  if (!data && new Date(data.exp * 1000) <= new Date()) {
+    await axios
```

Comment:
please use async/await syntax instead of promises syntax and especially try to avoid combining them in one statement as you do